### ISSUE TITLE
Internal compiler error with GCC 11.2 on macOS

### ISSUE THREAD

**Issue by microbit-carlos** [lancaster-university/microbit-v2-samples#47]
Released in February: https://developer.arm.com/downloads/-/arm-gnu-toolchain-downloads

```
[  4%] Building CXX object libraries/codal-core/CMakeFiles/codal-core.dir/source/core/codal_default_target_hal.cpp.obj
during GIMPLE pass: cddce
/Users/runner/work/microbit-v2-samples/microbit-v2-samples/libraries/codal-core/source/core/CodalUtil.cpp: In member function 'codal::KeyValueTableEntry* codal::KeyValueTable::find(uint32_t) const':
/Users/runner/work/microbit-v2-samples/microbit-v2-samples/libraries/codal-core/source/core/CodalUtil.cpp:65:1: internal compiler error: Illegal instruction: 4
   65 | }
      | ^
during GIMPLE pass: evrp
/Users/runner/work/microbit-v2-samples/microbit-v2-samples/libraries/codal-core/source/core/CodalCompat.cpp: In function 'int codal::string_reverse(char*)':
/Users/runner/work/microbit-v2-samples/microbit-v2-samples/libraries/codal-core/source/core/CodalCompat.cpp:149:1: internal compiler error: Illegal instruction: 4
  149 | }
      | ^
Please submit a full bug report,
with preprocessed source if appropriate.
See <https://bugs.linaro.org/> for instructions.
Please submit a full bug report,
with preprocessed source if appropriate.
See <https://bugs.linaro.org/> for instructions.
make[2]: *** [libraries/codal-core/CMakeFiles/codal-core.dir/source/core/CodalCompat.cpp.obj] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [libraries/codal-core/CMakeFiles/codal-core.dir/source/core/CodalUtil.cpp.obj] Error 1
```

```
during GIMPLE pass: evrp
/Users/runner/work/microbit-v2-samples/microbit-v2-samples/libraries/codal-core/source/core/CodalDmesg.cpp: In function 'void writeNum(char*, uint32_t, bool)':
/Users/runner/work/microbit-v2-samples/microbit-v2-samples/libraries/codal-core/source/core/CodalDmesg.cpp:181:1: internal compiler error: Illegal instruction: 4
  181 | }
      | ^
during GIMPLE pass: evrp
/Users/runner/work/microbit-v2-samples/microbit-v2-samples/libraries/codal-core/source/core/CodalComponent.cpp: In function 'void component_callback(codal::Event)':
/Users/runner/work/microbit-v2-samples/microbit-v2-samples/libraries/codal-core/source/core/CodalComponent.cpp:186:1: internal compiler error: Illegal instruction: 4
  186 | }
      | ^
Please submit a full bug report,
with preprocessed source if appropriate.
See <https://bugs.linaro.org/> for instructions.
make[2]: *** [libraries/codal-core/CMakeFiles/codal-core.dir/source/core/CodalComponent.cpp.obj] Error 1
Please submit a full bug report,
with preprocessed source if appropriate.
See <https://bugs.linaro.org/> for instructions.
make[2]: *** [libraries/codal-core/CMakeFiles/codal-core.dir/source/core/CodalDmesg.cpp.obj] Error 1
make[1]: *** [libraries/codal-core/CMakeFiles/codal-core.dir/all] Error 2
make: *** [all] Error 2
```
https://github.com/lancaster-university/microbit-v2-samples/runs/6311392236?check_suite_focus=true

**Comment by microbit-carlos** [comment]
This `internal compiler error: Illegal instruction: 4` looks like a confirmed bug in the compiler:
- https://community.arm.com/support-forums/f/compilers-and-libraries-forum/52623/gcc-11-2-arm-none-eabi-internal-compiler-error-illegal-instruction
- https://bugs.linaro.org/show_bug.cgi?id=5825

It's meant to be fixed in the next release that went out a couple of days ago.

**Comment by microbit-carlos** [comment]
I'll close this one as it is confirmed as a compiler bug that has been resolved in Arm gcc 11.2.
